user-guide/modules/ROOT/pages/advanced-builds.adoc

```diff
@@ -1,3 +1,247 @@
 = Advanced Builds
+:navtitle: Advanced Builds
+
+This section covers a number of advanced build scenarios that you may come across if your project has complex, legacy, or non-standard, components or processes.
+
+== Build with a Non-default Compiler
+
+Using the `b2` build system, you can specify the compiler you want to use with the `toolset` option. For example, if you want to use the GNU Compiler Collection (GCC), you would specify `toolset=gcc`. If you want to use Clang, you would use `toolset=clang`. For example, to build the Boost libraries with gcc:
+
+[source,txt]
+----
+./b2 toolset=gcc
+----
+
+If you want to use a specific _version_ of a compiler, you can specify it after the compiler name, separated by a dash. For example, to use GCC 7.3, you would use `toolset=gcc-7.3`:
+
+[source,txt]
+----
+./b2 toolset=gcc-7.3
+----
+
+You can also specify the toolset option when you're building your application. This is done in a similar way, by passing `toolset=` to the `b2` command, as part of your build process. Here's an example:
+
+[source,txt]
+----
+b2 toolset=clang
+----
+
+If the compiler is not in your system's PATH, you will need to provide the full path to the compiler executable. If there are spaces in the path, then wrap it in quotes, for example:
+
+[source,txt]
+----
+b2 toolset="C:\Program Files (x86)\Clang\"
+----
+
+== Specify the Application Binary Interface
+
+Building Boost with the old Application Binary Interface (ABI)) refers to compiling the library with a flag to ensure backward compatibility with an older version of an ABI.
+
+The ABI of a system is a specification that defines requirements for binary compatibility between applications and libraries that are built separately. It includes aspects such as calling conventions, byte order, and data structure alignment.
+
+In the context of pass:[C++], this generally comes up with GCC 5.x and later versions, which introduced a new ABI to conform more closely with the pass:[C++11] and pass:[C++14] standards. This new ABI changed the implementations of `std::string` and `std::list`, among other things, which broke binary compatibility with older versions of these libraries. This means that code compiled with the new ABI may _not_ be binary-compatible with code compiled with the old ABI.
+
+To build Boost with the old ABI when using GCC, you'd pass in the `-D_GLIBCXX_USE_CXX11_ABI=0` flag. For example:
```

Comment:
"Boost can be built with the old ABI using the `-D_GLIBCXX_USE_CXX11_ABI=0` flag:"

Since we already explain toolsets earlier, and because libstdc++ can be used with other compilers, I don't think its necessary to specify which particular compiler is used.

Reply:
ok